apps/remix-ide/src/app/panels/file-panel.js

```diff
@@ -103,194 +76,51 @@ module.exports = class Filepanel extends ViewPlugin {
-            if (Object.keys(filesList).length > 0) {
-              const workspacePath = Object.keys(filesList)[0].split('/').filter(val => val)
-              const workspaceName = workspacePath[workspacePath.length - 1]
-
-              this._deps.fileProviders.workspace.setWorkspace(workspaceName)
-              return resolve(workspaceName)
-            }
-            return reject(new Error('Can\'t find available workspace.'))
-          })
-        }
-      })
-    })
+  setWorkspaces (workspaces) {
+    this.worspaces = workspaces
   }
 
   async createNewFile () {
-    return await this.request.createNewFile()
-  }
+    const provider = this.fileManager.currentFileProvider()
+    const dir = provider.workspace || '/'
 
-  async uploadFile (event) {
-    return await this.request.uploadFile(event)
+    this.emit('displayNewFileInput', dir)
   }
 
-  async processCreateWorkspace (name) {
-    const workspaceProvider = this._deps.fileProviders.workspace
-    const browserProvider = this._deps.fileProviders.browser
-    const workspacePath = 'browser/' + workspaceProvider.workspacesPath + '/' + name
-    const workspaceRootPath = 'browser/' + workspaceProvider.workspacesPath
-    const workspaceRootPathExists = await browserProvider.exists(workspaceRootPath)
-    const workspacePathExists = await browserProvider.exists(workspacePath)
+  async uploadFile (target) {
+    const provider = this.fileManager.currentFileProvider()
+    const dir = provider.workspace || '/'
 
-    if (!workspaceRootPathExists) browserProvider.createDir(workspaceRootPath)
-    if (!workspacePathExists) browserProvider.createDir(workspacePath)
+    return this.emit('uploadFileEvent', dir, target)
   }
 
-  async workspaceExists (name) {
-    const workspaceProvider = this._deps.fileProviders.workspace
-    const browserProvider = this._deps.fileProviders.browser
-    const workspacePath = 'browser/' + workspaceProvider.workspacesPath + '/' + name
-    return browserProvider.exists(workspacePath)
-  }
-
-  async createWorkspace (workspaceName, setDefaults = true) {
-    if (!workspaceName) throw new Error('name cannot be empty')
-    if (checkSpecialChars(workspaceName) || checkSlash(workspaceName)) throw new Error('special characters are not allowed')
-    if (await this.workspaceExists(workspaceName)) throw new Error('workspace already exists')
-    else {
-      const workspaceProvider = this._deps.fileProviders.workspace
-      await this.processCreateWorkspace(workspaceName)
-      workspaceProvider.setWorkspace(workspaceName)
-      await this.request.setWorkspace(workspaceName) // tells the react component to switch to that workspace
-      if (setDefaults) {
-        for (const file in examples) {
-          try {
-            await workspaceProvider.set(examples[file].name, examples[file].content)
-          } catch (error) {
-            console.error(error)
-          }
-        }
-      }
-    }
+  async createWorkspace (workspaceName) {
```

Comment:
createWorkspace is an API call, it should return async response when the creation is done right? All file operations should be async.
Aside from that, dGIT uses the call to create an empty workspace with a bool flag. Any way this can be incorporated? deploying this will break dgit.